src/Libraries/Microsoft.Extensions.Http.AutoClient/AutoClientUtilities.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+
+namespace Microsoft.Extensions.Http.AutoClient;
+
+/// <summary>
+/// Utilities for AutoClient feature.
+/// </summary>
+public static class AutoClientUtilities
+{
+    private static readonly char[] _slashes = { '/', '\\' };
+    private static readonly char[] _slashesOrDot = { '/', '\\', '.' };
+
+    /// <summary>
+    /// Returns whether a value is a valid path argument.
+    /// </summary>
+    /// <param name="value">The value to validate.</param>
+    /// <returns>Whether the value is a valid path argument.</returns>
+    public static bool IsPathParameterValid(string value)
+    {
+        var trimmed = value.AsSpan().Trim();
+
+        if (trimmed.Length == 0)
+        {
+            return false;
+        }
+
+        // Fast path for most common cases
+        if (trimmed.IndexOfAny(_slashesOrDot) == -1)
+        {
+            return true;
+        }
+
+        // Slashes can't be used
+        if (trimmed.IndexOfAny(_slashes) >= 0)
+        {
+            return false;
+        }
+
```

Comment:
```suggestion
        // Fast path for most common cases
        var index = trimmed.IndexOfAny(_slashesOrDot);
        if (index == -1)
        {
            return true;
        }

        // Slashes can't be used
        trimmed = trimmed.Slice(index);
        if (trimmed.IndexOfAny(_slashes) >= 0)
        {
            return false;
        }

```

Reply:
Had to adapt the suggestion a little so that `trimmed` is not changed or it would have side effects with the last check on dots